### DISCUSSION TITLE
Copilot isnt really boosting my productivity - what am I doing wrong?

### DISCUSSION THREAD

**Discussion by MajorPainTheCactus** [community/community#51745] [Copilot Conversations]
### Select Topic Area

Question

### Body

Hi Im a C++ and HLSL coder and have been using Copilot for about a month but have found little benefit to using it. 

Why it isnt of benefit seems to be three fold: a) it relatively rarely suggests code, b) when it does I invariably have already pressed a key and the code suggestion vanishes and a lot of the time doesnt reappear, c) the code suggestion is invariably wrong.  

Yes the code it suggests compiles a lot of the time but its just never is really the code I want written. In the month of use (normally most days for a good few hours) its probably written code Ive used that I can count on the fingers of one hand. However during that time Ive had to read and delete/modify a lot of wrong/redundant suggested code. Admittedly because it so rarely suggests its not a huge issue but Im certainly not getting large productivity gains.

What am I doing wrong? Id have thought it would just work out of the box. I do think a lot of the time it would need to read my mind to generate the correct code so maybe my expectations are too high. Im using VS2022 btw.

**Comment by ayim** [comment]
Hey! PM for Copilot in VS here - does writing comments help?
Recently, we've been aggressively dogfooding Copilot for C++ to understand model quality & utility... definitely an area we're trying to explore and improve.
